src/commands/auth/accesstoken/store.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as os from 'os';
+import { flags, FlagsConfig, SfdxCommand } from '@salesforce/command';
+import { AuthFields, AuthInfo, AuthInfoConfig, Messages, sfdc, SfdxError } from '@salesforce/core';
+import { ensureString, getString } from '@salesforce/ts-types';
+import { env } from '@salesforce/kit';
+import { Prompts } from '../../../prompts';
+import { Common } from '../../../common';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/plugin-auth', 'accesstoken.store');
+const commonMessages = Messages.loadMessages('@salesforce/plugin-auth', 'messages');
+
+const ACCESS_TOKEN_FORMAT1 = '"<org id>!<accesstoken>"';
+
+export default class Store extends SfdxCommand {
+  public static readonly description = messages.getMessage('description', [ACCESS_TOKEN_FORMAT1]);
+  public static readonly examples = messages.getMessage('examples').split(os.EOL);
+  public static aliases = ['force:auth:accesstoken:store'];
+
+  public static readonly flagsConfig: FlagsConfig = {
+    instanceurl: flags.url({
+      char: 'r',
+      description: commonMessages.getMessage('instanceUrl'),
+      required: true,
+    }),
+    setdefaultdevhubusername: flags.boolean({
+      char: 's',
+      description: commonMessages.getMessage('setDefaultDevHub'),
+      default: false,
+    }),
+    setdefaultusername: flags.boolean({
+      char: 's',
+      description: commonMessages.getMessage('setDefaultUsername'),
+      default: false,
+    }),
+    setalias: flags.string({
+      char: 'a',
+      description: commonMessages.getMessage('setAlias'),
+    }),
+    noprompt: flags.boolean({
+      char: 'p',
+      description: commonMessages.getMessage('noPrompt'),
+      required: false,
+      default: false,
+    }),
+  };
+
+  public async run(): Promise<AuthFields> {
+    const instanceUrl = ensureString(getString(this.flags, 'instanceurl.href'));
+    const accessToken = await this.getAccessToken();
+    const authInfo = await this.getUserInfo(accessToken, instanceUrl);
+    return await this.storeAuthFromAccessToken(authInfo);
```

Comment:
Outside of try/catch blocks, `return await` is redundant. There's even an ESLint rule to detect it, but it allows it in try/catch.